dashboard/factories.py

```diff
@@ -38,9 +38,8 @@ class CachedCertificateFactory(DjangoModelFactory):
         "certificate_type": "verified",
         "grade": randint(60, 100)/100.0,
         "course_id": x.course_run.edx_course_key,
+        "username": x.user.username,
```

Comment:
the problem I have is that I need that the username in the cached data is consistent, because some code (edx_api_client.CurrentGrades) checks that for obvious reasons.
Now I really do not care what we put here, but I guess that in the cases you are talking about the solution is to explicitly pass a custom defined `data` parameter for the factory.